src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksNewVendorRequestsServiceImpl.java

```diff
@@ -195,49 +195,15 @@ private boolean validateVendorType(String vendorType, List<String> errorMessages
-            }
-        }
-        return allValidationPassed;
-    }
-    
-    private boolean combinedLegalFirstLastNameEnteredConformsToMaxKfsLength(PaymentWorksVendor stgNewVendorRequestDetailToProcess, List<String> errorMessages) {
-        boolean allValidationPassed = true;
-        int lastDelimiterFirstCombinedLength = stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalLastName().length() + 
-                                               VendorConstants.NAME_DELIM.length() + 
-                                               stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalFirstName().length();
-
-        if (lastDelimiterFirstCombinedLength > VendorConstants.MAX_VENDOR_NAME_LENGTH) {
-            errorMessages.add(getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.ERROR_COMBINED_LEGAL_FIRST_LAST_NAME_TOO_LONG_FOR_KFS));
-            allValidationPassed = false;
-            LOG.info("isCombinedLegalFirstLastNameEnteredWithinProperLength: PMW Legal Last Name length is " + stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalLastName().length() +
-                     " PMW Legal First Name length is " + stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalFirstName().length() +
-                     " which when combined with delimiters must be less that the KFS legal name max length of " + (VendorConstants.MAX_VENDOR_NAME_LENGTH - VendorConstants.NAME_DELIM.length()));
-        }
-        return allValidationPassed;
-    }
-    
-    private boolean companyLegalNameEnteredConformsToMaxKfsLength(PaymentWorksVendor stgNewVendorRequestDetailToProcess, List<String> errorMessages) {
-        boolean allValidationPassed = true;
-        if (stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalName().length() > VendorConstants.MAX_VENDOR_NAME_LENGTH) {
-            errorMessages.add(getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.ERROR_LEGAL_NAME_TOO_LONG_FOR_KFS));
-            allValidationPassed = false;
-            LOG.info("companyLegalNameEnteredConformsToMaxKfsLength: KFS legal name max length is " + VendorConstants.MAX_VENDOR_NAME_LENGTH +
-                     " PMW legal name length received is " + stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalName().length());
+    private boolean legalNameWasEntered(PaymentWorksVendor stgNewVendorRequestDetailToProcess, List<String> errorMessages) {
+        if ( (StringUtils.isNotBlank(stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalLastName()) &
+              StringUtils.isNotBlank(stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalFirstName()))
+              || (StringUtils.isNotBlank(stgNewVendorRequestDetailToProcess.getRequestingCompanyLegalName())) ) {
+            return true;
+        } else {
+            errorMessages.add(getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.ERROR_LEGAL_NAME_NULL_OR_BLANK));
```

Comment:
This error message is for legal name missing or first & last name missing.  Should it be renamed so it mentions that?

Reply:
Legal name is considered to be one thing by the database or the business rules (depending upon the individual or business tax reporting) and is composed later on in the code. This is a hard error that is done up front that will require the Purchasing Staff to address the problem in PMW and will stop a PVEN edoc from being created (force hand entry into KFS of the data).  Therefore; the Purchasing staff will actually see which type of legal name it is in PMW because one-and-only-one legal name is allowed (EIN only presents business legal name and SSN presents first last name) so IMHO getting that granular here may add more confusion.

Reply:
OK, fair enough